### PR TITLE
Rb193 enc params fixes

### PR DESCRIPTION
**Pull request by domcleal**
Two Ruby 1.9 fixes that seem to break the ENC when using class parameters:

<pre>
undefined method `each' for "fqdn":String

NoMethodError
undefined method `each' for "fqdn":String
app/models/classification.rb:87:in `block in path2matches'
app/models/classification.rb:85:in `each'
app/models/classification.rb:85:in `path2matches'
app/models/classification.rb:55:in `values_hash'
app/models/classification.rb:12:in `enc'
app/models/host.rb:724:in `lookup_keys_class_params'
app/models/host.rb:353:in `info'
app/controllers/hosts_controller.rb:166:in `block (2 levels) in externalNodes'
app/controllers/hosts_controller.rb:165:in `externalNodes'9
lib/foreman/threadsession.rb:31:in `clear_thread' 
</pre>

<pre>
private method `path_elements' called for #<LookupKey:0x00000003d9bec8> 

NoMethodError
private method `path_elements' called for #<LookupKey:0x00000003d9bec8>
app/models/classification.rb:47:in `map'
app/models/classification.rb:47:in `possible_value_orders'
app/models/classification.rb:85:in `path2matches'
app/models/classification.rb:55:in `values_hash'
app/models/classification.rb:12:in `enc'
app/models/host.rb:724:in `lookup_keys_class_params'
app/models/host.rb:353:in `info'
app/controllers/hosts_controller.rb:166:in `block (2 levels) in externalNodes'
app/controllers/hosts_controller.rb:165:in `externalNodes'
lib/foreman/threadsession.rb:31:in `clear_thread' 
</pre>

### CODE REVIEW

File: app/models/lookup_key.rb [hunk 1]
```diff
@@ -142,14 +150,6 @@ def attr_to_value host, element
...
   end
 
-  def path_elements
-    path.split.map do |paths|
-      paths.split(KEY_DELM).map do |element|
-        element
-      end
-    end
-  end
-
```
Comment (GregSutcliffe) [inline app/models/lookup_key.rb:152]: This look identical, just moved up 40 lines. Necessary?

Comment (domcleal) [inline app/models/lookup_key.rb:152]: Yes, it's been moved above the "private" line.  Seemed neater than explicitly declaring it public.

Comment (GregSutcliffe) [inline app/models/lookup_key.rb:152]: Cool.

**Comment by GregSutcliffe** [comment]
merged in a4008a128a5675228f47f1794b98df5b753d78f3. Thanks!

**Pull request closed without merging**